scripts/serial_link.py

```diff
@@ -382,6 +382,30 @@ def format_log_json_entry(t0, item):
   data = item.to_json_dict()
   return {'delta': delta, 'timestamp': timestamp, 'data': data}
 
+class Watchdog:
+  def __init__(self, timeout, alarm):
+    self.timeout = timeout
+    self.alarm = alarm
+    self.timer = None
+
+  def __call__(self, *args):
+    self.call()
+
+  def call(self):
+    if self.timer:
+      self.timer.cancel()
+    self.timer = threading.Timer(self.timeout, self.alarm)
```

Comment:
Also, lame question, but does `self.timer` need to be on a daemon thread (i.e., `self.timer.daemon = True`?